devicetypes/smartthings/zigbee-lock.src/zigbee-lock.groovy

```diff
@@ -243,9 +243,9 @@ def reloadAllCodes() {
 	sendEvent(lockCodesEvent(lockCodes))
 	def cmds = validateAttributes()
 	if (isYaleLock()) {
-		state.checkCode = 1
+		state.checkCode = state.checkCode ?: 1
```

Comment:
is this supposed to be a null check? Elvis operators treat 0 as false too. And if the value is anything other than 0/null/false, it will keep that value instead of setting the value to 1. is that the intended behavior?

Reply:
Yes this is the intended behavior. 0 slot is not valid for Yale locks hence 1 should be picked up in that case. For Kwikset locks 0 is a valid slot however even if elvis operator considers it as false we are picking up 0 by default at line 248.